### ISSUE TITLE
CDM16 Sample trace SUBJECT_PROCESS uuid re-used as FileObject.   

### ISSUE THREAD

**Issue by tedhabeck** [cadets/trace-data#2]
BasicOps.cdm.bin/BasicOps.cdm.json contains re-purposed UUID.   Is this expected?

```bash

Origional datatype of UUID: 8d41cee4-f8f8-11e6-a0ed-44a8421f8dc6 was: SUBJECT_PROCESS new type is: FileObject
Origional datatype of UUID: 8de05ebf-f8f8-11e6-a0ed-44a8421f8dc6 was: SUBJECT_PROCESS new type is: FileObject
Origional datatype of UUID: 8de5d0c4-f8f8-11e6-a0ed-44a8421f8dc6 was: SUBJECT_PROCESS new type is: FileObject
```

**Comment by arunthomas** [comment]
Thanks for the report, Ted. This looks like a bug. We'll dig into the issue.

**Comment by amstrnad** [comment]
How interesting! The second object should not have been created. I'll get an updated trace out later this afternoon that will not redefine the UUID (and should include some missing fork events as well).

The CDM translator was making an assumption that anything passed to `close` was some sort of file, so it created a file object. In actuality, it is possible to close a process descriptor as well as a file descriptor.

**Comment by amstrnad** [comment]
Okay, that should be fixed, though I think some of the behavior in the updated trace may be surprising to analysts.

FreeBSD has a syscall that returns a process descriptor ([pdfork](https://www.freebsd.org/cgi/man.cgi?query=pdfork&sektion=2)). This process descriptor can be used in a few other syscalls that normally take file descriptors (primarily `close`).

**Comment by amstrnad** [comment]
Whoops, actually fixed in cbc429c. Sorry about that.
